lib/IRGen/GenClangDecl.cpp

```diff
@@ -333,3 +339,15 @@ void IRGenModule::finalizeClangCodeGen() {
   ClangCodeGen->HandleTranslationUnit(
       *const_cast<clang::ASTContext *>(ClangASTContext));
 }
+
+void IRGenModule::ensureImplicitCXXDestructorBodyIsDefined(
+    clang::CXXDestructorDecl *destructor) {
+  if (!destructor->isUserProvided() &&
+      !destructor->doesThisDeclarationHaveABody()) {
+    assert(!destructor->isDeleted() &&
+           "Swift cannot handle a type with no known destructor.");
```

Comment:
Would this assertion fail if `struct Base` has a deleted destructor but `struct Derived : Base` defines an explicit destructor? As far as I can see we would currently import `Derived` into Swift, and then fail with an assertion here.

Reply:
Good question. Let me test!

Reply:
Yep, this was triggered. I fixed it by adding a check for a deleted destructor before the new invocation path, and added a test too.